### PR TITLE
#56215 - Introduce the `is_term_publicly_viewable()` function

### PR DESCRIPTION
**Pull request by johnbillion**
Trac ticket: https://core.trac.wordpress.org/ticket/56215

### CODE REVIEW

File: tests/phpunit/tests/term/isTermPubliclyViewable.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	/**
+	 * Unit tests for is_term_publicly_viewable().
+	 *
+	 * @ticket 56215
+	 */
+	public function test_non_existent_term_is_not_publicly_viewable() {
+		$this->assertFalse( is_term_publicly_viewable( 123 ) );
+	}
+
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/term/isTermPubliclyViewable.php]: Are you able to move this above `test_is_term_publicly_viewable()`? I think convention is to keep the test and data methods consecutive whenever possible.

**Review by peterwilsoncc — COMMENTED** [review]
I think this is a good addition as a sibling to `is_post_publicly_viewable()`.

LGTM but a reordering request in the tests.

File: tests/phpunit/tests/term/isTermPubliclyViewable.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @group term
```
Comment (costdev) [inline tests/phpunit/tests/term/isTermPubliclyViewable.php:4]:
```suggestion
 * @group term
 *
 * @covers ::is_term_publicly_viewable
```

File: tests/phpunit/tests/term/isTermPubliclyViewable.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	/**
+	 * Unit tests for is_term_publicly_viewable().
+	 *
+	 * @dataProvider data_is_term_publicly_viewable
+	 * @ticket 56215
+	 *
+	 * @param string $taxonomy The taxonomy name.
+	 * @param bool   $expected The expected result of the function call.
+	 */
+	public function test_is_term_publicly_viewable( $taxonomy, $expected, $parent_key = '' ) {
```
Comment (costdev) [inline tests/phpunit/tests/term/isTermPubliclyViewable.php]: `$parent_key` isn't documented or used in the test method.

**Review by costdev — CHANGES_REQUESTED** [review]

**Comment by johnbillion** [comment]
https://core.trac.wordpress.org/changeset/53893

**Pull request closed without merging**